architecture-model-generator/modules/architecture-model-generator-core/src/main/java/io/ballerina/artifactsgenerator/ModuleNodeTransformer.java

```diff
@@ -0,0 +1,246 @@
+                        .icon(connection.get());
+            } else {
+                variableBuilder.type(Artifact.Type.VARIABLE);
+            }
+        }
+
+        return Optional.of(variableBuilder.build());
+    }
+
+    @Override
+    public Optional<Artifact> transform(TypeDefinitionNode typeDefinitionNode) {
+        Artifact.Builder typeBuilder = new Artifact.Builder(typeDefinitionNode)
+                .name(typeDefinitionNode.typeName().text())
+                .type(Artifact.Type.TYPE);
+        return Optional.of(typeBuilder.build());
+    }
+
+    @Override
+    protected Optional<Artifact> transformSyntaxNode(Node node) {
+        return Optional.empty();
+    }
+
+    private void setIcon(Artifact.Builder builder, Node node) {
+        Optional<TypeSymbol> typeSymbol = semanticModel.typeOf(node);
+        if (typeSymbol.isEmpty()) {
+            return;
+        }
+        if (typeSymbol.get().typeKind() == TypeDescKind.UNION) {
+            UnionTypeSymbol unionTypeSymbol = (UnionTypeSymbol) typeSymbol.get();
+            Optional<TypeSymbol> listenerSymbol = unionTypeSymbol.memberTypeDescriptors().stream()
+                    .filter(member -> !member.subtypeOf(semanticModel.types().ERROR))
+                    .findFirst();
+            listenerSymbol.ifPresent(builder::icon);
+            return;
+        }
+        builder.icon(typeSymbol.get());
+    }
+
+    private Optional<ClassSymbol> getConnection(Node node) {
+        try {
+            Symbol symbol = semanticModel.symbol(node).orElseThrow();
+            TypeReferenceTypeSymbol typeDescriptorSymbol =
+                    (TypeReferenceTypeSymbol) ((VariableSymbol) symbol).typeDescriptor();
+            ClassSymbol classSymbol = (ClassSymbol) typeDescriptorSymbol.typeDescriptor();
+            if (classSymbol.qualifiers().contains(Qualifier.CLIENT)) {
+                return Optional.of(classSymbol);
+            }
+        } catch (Throwable e) {
+            // Ignore
+        }
+        return Optional.empty();
+    }
+
+    private static String getPathString(NodeList<Node> nodes) {
+        return nodes.stream()
+                .map(node -> node.toString().trim())
+                .collect(Collectors.joining());
+    }
+
+    private static boolean hasQualifier(NodeList<Token> qualifierList, SyntaxKind kind) {
+        return qualifierList.stream().anyMatch(qualifier -> qualifier.kind() == kind);
+    }
+
+    private boolean isPromptAsCodeFunction(FunctionDefinitionNode functionDefinitionNode) {
+        Optional<Symbol> funcSymbol = this.semanticModel.symbol(functionDefinitionNode);
+        if (funcSymbol.isEmpty() || !((FunctionSymbol) funcSymbol.get()).external()) {
+            return false;
+        }
+
+        List<AnnotationAttachmentSymbol> annotAttachments = ((ExternalFunctionSymbol) funcSymbol.get())
+                .annotAttachmentsOnExternal();
+        return annotAttachments.stream()
```

Comment:
We don't use these `@np` annotations any more.

Reply:
I am using the existing implementation of the /components API. We need a separate PR to address this change

Reply:
Yes, we can do this later. They are introduce a new ballerina syntax.